File: platform/packages_mjs/env-resolver/src/env-jira.x.ts
import { resolve } from '@internal/env-resolve';

export interface JiraEnv {
    baseUrl: string | undefined;
    email: string | undefined;
    apiToken: string | undefined;
}

export function resolveJiraEnv(config?: Record<string, any>): JiraEnv {
    return {
        baseUrl: resolve(undefined, ['JIRA_BASE_URL'], config, 'baseUrl', undefined),
        email: resolve(undefined, ['JIRA_EMAIL'], config, 'email', undefined),
        apiToken: resolve(undefined, ['JIRA_API_TOKEN'], config, 'apiToken', undefined),
    };
}

File: platform/packages_py/env_resolver/env_resolver/env_jira_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve


@dataclass(frozen=True)
class JiraEnv:
    base_url: str | None
    email: str | None
    api_token: str | None


def resolve_jira_env(config: dict[str, Any] | None = None) -> JiraEnv:
    return JiraEnv(
        base_url=resolve(None, ['JIRA_BASE_URL'], config, 'base_url', None),
        email=resolve(None, ['JIRA_EMAIL'], config, 'email', None),
        api_token=resolve(None, ['JIRA_API_TOKEN'], config, 'api_token', None),
    )
